tests/test_views.py

```diff
@@ -1,85 +1,82 @@
-
-    def test_view_anonymous_user(self):
+    def test_anonymous_user_cannot_access_preview(self):
         response = self.client.get(self.preview_url)
         expected_url = "/en/admin/login/?next=" + self.preview_url
         self.assertEqual(response.status_code, 302)
         self.assertEqual(response.url, expected_url)
 
-    def test_view_standard_user(self):
+    def test_standard_user_cannot_access_preview(self):
         standard_user = self.get_standard_user()
         with self.login_user_context(standard_user):
             response = self.client.get(self.preview_url)
             expected_url = "/en/admin/login/?next=" + self.preview_url
             self.assertEqual(response.status_code, 302)
             self.assertEqual(response.url, expected_url)
 
-    def test_view_super_user(self):
-        staff_user = self.get_superuser()
-        with self.login_user_context(staff_user):
-            response = self.client.get(self.preview_url)
-            self.assertEqual(response.status_code, 200)
+
+class PreviewViewTestCases(CMSTestCase):
+    def setUp(self):
+        self.menu_content = MenuContentFactory()
+        self.preview_url = admin_reverse(
+            "djangocms_navigation_menuitem_preview",
+            kwargs={"menu_content_id": self.menu_content.id},
+        )
+
+    def test_view_url(self):
+        expected_url = "/en/admin/djangocms_navigation/menuitem/{}/preview/".format(
+            self.menu_content.id
+        )
+        self.assertEqual(self.preview_url, expected_url)
 
     def test_view_context_data_with_valid_menu_content(self):
-        factory = RequestFactory()
-        request = factory.get(self.preview_url)
-        request.user = self.get_superuser()
-        view = MenuContentPreviewView()
-        view.request = request
-        view.kwargs = {"menu_content_id": self.menu_content.id}
-        response = view.get_context_data()
-        self.assertIsInstance(response["view"], MenuContentPreviewView)
-        self.assertIn("annotated_list", response)
-
-    def test_view_context_data_with_invalid_int_menu_content(self):
-        factory = RequestFactory()
-        request = factory.get(self.preview_url)
-        request.user = self.get_superuser()
-        view = MenuContentPreviewView()
-        view.request = request
-        view.kwargs = {"menu_content_id": 99}
-        response = view.get_context_data()
-        self.assertEqual(response.status_code, 400)
-
-    def test_view_context_data_with_invalid_string_menu_content(self):
-        factory = RequestFactory()
-        request = factory.get(self.preview_url)
-        request.user = self.get_superuser()
-        view = MenuContentPreviewView()
-        view.request = request
-        view.kwargs = {"menu_content_id": "dummy"}
-        response = view.get_context_data()
-        self.assertEqual(response.status_code, 400)
+        super_user = self.get_superuser()
+        with self.login_user_context(super_user):
```

Comment:
Now that you've divided this up into two test cases, you can log in the super user in `setUp`. Saves you having to write these two lines in every test.